src/lib/matrix/matrix/Slice.hpp

```diff
@@ -332,4 +332,22 @@ class Slice
 	Matrix<Type, M, N> *_data;
 };
 
+/**
+ * + operator for supporting 'Slice + Vector' addition for any dimension : e.g. Vector2, Vector3, ...
+ *
+ * The 'Vector + Slice' case (e.g. Vector2f + Vector3f.xy()) has been working due to the Slice object (Vector3f.xy())
+ * being implicitely transformed into a Vector2f through the Vector2 constructor. However, there were errors when you
+ * tried to do 'Slice + Vector' since there is no Slice constructor out of Vector object.
+ *
+ * This is why this special 'addition' operator support is needed, to allow addition in any order.
```

Comment:
I think you should mention this in the commit but not here, it's just too much if we need to justify each operator overload with a paragraph of text. If someone has a question, he can still git blame to find why this has been added.

Reply:
Never thought about putting it in the git commit, that is a good idea!